SI-10240 Better pos for infix type arg

som-snytt opened the pull request:
The range position for the type apply in `1 ->[Int] 2`
should include the last type arg.

It would be nice to include the right bracket.

som-snytt committed:
SI-10240 Better pos for infix type arg

The range position for the type apply in `1 ->[Int] 2`
should include the last type arg.

It would be nice to include the right bracket.

adriaanm commented on src/compiler/scala/tools/nsc/ast/parser/Parsers.scala:
```diff
@@ -824,7 +824,10 @@ self =>
         def sel = atPos(opPos union t.pos)(Select(stripParens(t), op.encode))
-        if (targs.isEmpty) sel else atPos(left.pos)(TypeApply(sel, targs))
+        if (targs.isEmpty) sel
+        else atPos(sel.pos union targs.last.pos withPoint sel.pos.point) {
```
`sel` is a `def` (for `def`ensive tree duplication?) I suggest turning it into a `val`, along with its position, so that both are evaluated once

som-snytt replied:
Good eyes! I guess those spectacles are a recent prescription.

som-snytt replied:
sorry, next time!

som-snytt replied:
The follow-up was the second commit, this diff was outdated.

som-snytt committed:
SI-10240 Additional scouting

Val for pos and tree, thanks Mr Moors.

Also changed a long line to the boolean style you see quite
a lot in the code base.

som-snytt commented:
I wish there was a beer called `Mr Moors`.
